opamp/observiq/observiq_packages_state_provider.go

```diff
@@ -156,7 +172,9 @@ func (p *packagesStateProvider) SetLastReportedStatuses(statuses *protobufs.Pack
 		p.logger.Debug("Failed to delete package statuses yaml", zap.Error(err))
 	}
 
-	data, err := yaml.Marshal(statuses)
+	states := packageStatusesToStates(statuses)
+
+	data, err := yaml.Marshal(*states)
```

Comment:
Do you need to dereference the pointer here? If so maybe we shouldn't return a pointer from this method.